diff --git a/activity/activity.info b/activity/activity.info
index 57a2158..960a710 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,9 +2,18 @@
 name = falabracman
 activity_version = 2
 host_version = 1
-bundle_id = org.laptop.community.falabracman
+bundle_id = org.sugarlabs.falabracmanActivity
+release_date = 2020-03-18
 icon = activity
 license = GPLv3+
 exec = sugar-activity3 activity.Falabracman
 show_launcher = yes
 repository = https://github.com/sugarlabs/falabracman-activity.git
+metadata_license = CC0-1.0
+description = Help Falabracman learn some words, by collecting all the letters in the right order.
+summary = Collect the letters of the word in correct order
+update_contact = tch@sugarlabs.org
+tags = Game
+developer_name = Sugar Labs Community
+url = https://github.com/sugarlabs/falabracman-activity
+screenshots = https://activities.sugarlabs.org/en-US/sugar/images/p/273/1252990860
