diff --git a/constants.py b/constants.py
index 6253f38..0ba9986 100644
--- a/constants.py
+++ b/constants.py
@@ -26,6 +26,7 @@ gi.require_version("Gtk", "3.0")
 from gi.repository import Gtk
 from gi.repository import Gdk
 
+from sugarapp.helpers import PrimaryMonitor
 
 LOCAL_DIR = os.path.dirname(os.path.realpath(__file__))
 
@@ -33,7 +34,7 @@ DRAG_TARGETS = [Gtk.TargetEntry.new("BALL", Gtk.TargetFlags.SAME_APP, 0)]
 IGNORE_TARGETS = []
 DRAG_ACTION = Gdk.DragAction.MOVE
 
-BALL_SIZE = (Gdk.Screen.height() - 200) / 10
+BALL_SIZE = (PrimaryMonitor.height() - 200) / 10
 
 
 class BallType:
diff --git a/helpbutton.py b/helpbutton.py
index 74450ef..4672970 100644
--- a/helpbutton.py
+++ b/helpbutton.py
@@ -28,6 +28,7 @@ from gi.repository import Gdk
 
 from sugar3.graphics.toolbutton import ToolButton
 
+from sugarapp.helpers import PrimaryMonitor
 
 class HelpButton(Gtk.ToolItem):
 
@@ -41,10 +42,10 @@ class HelpButton(Gtk.ToolItem):
         self._palette = help_button.get_palette()
 
         sw = Gtk.ScrolledWindow()
-        sw.set_size_request(int(Gdk.Screen.width() / 2.8), 310)
+        sw.set_size_request(int(PrimaryMonitor.width() / 2.8), 310)
         sw.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
 
-        self._max_text_width = int(Gdk.Screen.width() / 3) - 600
+        self._max_text_width = int(PrimaryMonitor.width() / 3) - 600
         self._vbox = Gtk.Box()
         self._vbox.set_orientation(Gtk.Orientation.VERTICAL)
         self._vbox.set_homogeneous(False)
