diff --git a/activity.py b/activity.py
index 5eb236a..6d1d633 100755
--- a/activity.py
+++ b/activity.py
@@ -11,7 +11,6 @@ from gi.repository import GLib
 from gi.repository import Gdk
 from gi.repository import Gtk
 
-from sugar3.activity import activity
 from sugar3.presence.presenceservice import PresenceService
 from sugar3.activity.widgets import ActivityToolbarButton
 from sugar3.activity.widgets import StopButton
@@ -26,15 +25,17 @@ from gettext import gettext as _
 from textchannel import TextChannelWrapper
 import game
 
+from sugarapp.widgets import SugarCompatibleActivity
 
-class MazeActivity(activity.Activity):
+
+class MazeActivity(SugarCompatibleActivity):
 
     def __init__(self, handle):
         """Set up the Maze activity."""
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
         self._busy_count = 0
         self._unbusy_idle_sid = None
-
+        
         if 'state' in self.metadata:
             self.state = json.loads(self.metadata['state'])
         else:
