diff --git a/activity/activity.info b/activity/activity.info
index 609f93c..958854d 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -1,10 +1,19 @@
 [Activity]
 name = Napier
 activity_version = 5
-license = GPLv3
-bundle_id = org.sugarlabs.NapierActivity
+license = GPLv3+
+bundle_id = org.sugarlabs.Napier
 exec = sugar-activity3 NapierActivity.NapierActivity
 icon = activity-napier
 show_launcher = yes
 summary = Napier's bones are a multiplication calculator from the 16th century
 category = maths
+tags = Education
+release_date = 2020-03-20
+description = You can use Napier's Bones to multiple any multi-digit number by a single digit number (1-9).
+metadata_license = CC0-1.0
+update_contact = tch@sugarlabs.org
+developer_name = Sugar Labs Community
+repository = https://github.com/sugarlabs/napier.git
+url = https://activities.sugarlabs.org/en-US/sugar/addon/4429
+screenshots = https://i.imgur.com/c508M4A.png
