diff --git a/SlideruleActivity.py b/SlideruleActivity.py
index c167d04..5c2ed21 100644
--- a/SlideruleActivity.py
+++ b/SlideruleActivity.py
@@ -31,7 +31,7 @@ import gi
 gi.require_version('Gtk', '3.0')
 
 from gi.repository import Gtk, Gdk
-
+from sugarapp.widgets import SugarCompatibleActivity
 from sugar3.activity import activity
 from sugar3.graphics.toolbarbox import ToolbarBox
 from sugar3.bundle.activitybundle import ActivityBundle
@@ -69,12 +69,12 @@ UK_unknown = '-'
 FUNCTIONS = [FL_add, FC_multiply, FCI_divide, FA_square, FK_cube, FS_sin,
              FT_tan, FE_natural_log, UK_unknown]
 
-
-class SlideruleActivity(activity.Activity):
+showCopyPaste = False
+class SlideruleActivity(SugarCompatibleActivity):
     """ A sliderule activity for Sugar """
 
     def __init__(self, handle):
-        super(SlideruleActivity,self).__init__(handle)
+        super().__init__(handle)
 
         self._setup_toolbars()
 
@@ -473,12 +473,13 @@ class SlideruleActivity(activity.Activity):
         toolbox.toolbar.insert(custom_stator_toolbar_button, -1)
         custom_stator_toolbar_button.show()
 
-        edit_toolbar_button = ToolbarButton(label=_('Edit'),
-                                            page=edit_toolbar,
-                                            icon_name='toolbar-edit')
-        edit_toolbar_button.show()
-        toolbox.toolbar.insert(edit_toolbar_button, -1)
-        edit_toolbar_button.show()
+        if showCopyPaste:
+            edit_toolbar_button = ToolbarButton(label=_('Edit'),
+                                                page=edit_toolbar,
+                                                icon_name='toolbar-edit')
+            edit_toolbar_button.show()
+            toolbox.toolbar.insert(edit_toolbar_button, -1)
+            edit_toolbar_button.show()
 
         self.set_toolbar_box(toolbox)
         toolbox.show()
@@ -572,16 +573,17 @@ class SlideruleActivity(activity.Activity):
                     ENTRY_CALLBACK[i],
                     tooltip=ENTRY_TOOLTIP[i]))
 
-        copy = button_factory('edit-copy',
-                              edit_toolbar,
-                              self._copy_cb,
-                              tooltip=_('Copy'),
-                              accelerator='<Ctrl>c')
-        paste = button_factory('edit-paste',
-                               edit_toolbar,
-                               self._paste_cb,
-                               tooltip=_('Paste'),
-                               accelerator='<Ctrl>v')
+        if showCopyPaste:
+            copy = button_factory('edit-copy',
+                                edit_toolbar,
+                                self._copy_cb,
+                                tooltip=_('Copy'),
+                                accelerator='<Ctrl>c')
+            paste = button_factory('edit-paste',
+                                edit_toolbar,
+                                self._paste_cb,
+                                tooltip=_('Paste'),
+                                accelerator='<Ctrl>v')
 
         separator_factory(toolbox.toolbar, True, False)
 
@@ -590,6 +592,7 @@ class SlideruleActivity(activity.Activity):
         toolbox.toolbar.insert(stop_button, -1)
         stop_button.show()
         # workaround to #2050
-        edit_toolbar_button.set_expanded(True)
+        if showCopyPaste:
+            edit_toolbar_button.set_expanded(True)
         # start with project toolbar enabled
         project_toolbar_button.set_expanded(True)
